### ISSUE TITLE
[TASK] Port Filterquery with two Functions

### ISSUE THREAD

**Issue by keyseven123** [nebulastream/nebulastream#264]
### Description

After #102, #103, #104, and #263, we can port the filter operator to the new nautilus-lib. We plan on having a running filter query with two functions.

If we see that the functions registry takes too much time or pollutes the PR, we will create a new issue/pr for it
